src/game.html

```diff
@@ -10116,6 +10426,209 @@
             },
         ];
 
+        const roamingEnemies = {
+            enemies: [],
+            maxEnemies: 6,
+            nextId: 0,
+            enemiesKilledOnFloor: 0,
+            respawningEnabled: false,
+
+            // Check if a position is in a boulder's path
+            isInBoulderPath: function(x, y) {
+                if (!boulder.isActiveOnFloor) return false;
+                
+                if (boulder.direction === 'horizontal') {
+                    return y === boulder.y && x >= boulder.hallwayBounds.start.x && x <= boulder.hallwayBounds.end.x;
+                } else {
+                    return x === boulder.x && y >= boulder.hallwayBounds.start.y && y <= boulder.hallwayBounds.end.y;
+                }
```

Comment:
This could be shortened up a bit with a ternary if desired

```suggestion
                return boulder.direction === 'horizontal'
                    ? boulder.y && x >= boulder.hallwayBounds.start.x && x <= boulder.hallwayBounds.end.x
                    : boulder.x && y >= boulder.hallwayBounds.start.y && y <= boulder.hallwayBounds.end.y;
```